README.md
# Rypjaws VS Code Theme Extension

<p align="center">
<img src="assets/RypJaws.png" alt="Rypjaws VS Code Theme">
  <br>
  Rypjaws
</p>
Welcome to the Rypjaws VS Code Theme Extension! This extension brings the Rypjaws theme to your Visual Studio Code editor, providing a stylish and immersive coding experience.

## Features

- **Elegant Color Scheme:** Rypjaws theme offers a carefully chosen set of colors that are easy on the eyes and enhance your focus on the code.
- **Syntax Highlighting:** The theme provides syntax highlighting for various programming languages, making your code stand out and improving readability.
- **Consistent Design:** Rypjaws maintains a consistent design across different elements of the editor, ensuring a cohesive and visually pleasing experience.
- **Customized UI Components:** The extension modifies the appearance of various UI components, including the sidebar, activity bar, status bar, and more, to match the overall aesthetic of the theme.
- **Optimized for Productivity:** Rypjaws is designed to help you stay productive with its carefully selected colors and contrast ratios, allowing you to work for long hours without straining your eyes.

## Installation

1. Launch Visual Studio Code.
2. Open the Extensions view by clicking on the square icon in the left sidebar or by pressing `Ctrl+Shift+X` (`Cmd+Shift+X` on macOS).
3. Search for "Rypjaws" in the Extensions Marketplace.
4. Click on the "Install" button for the Rypjaws VS Code Theme Extension.
5. Once installed, click on the "Reload" button to activate the theme.

## Activation

To activate the Rypjaws theme, follow these steps:

1. Open the Command Palette by pressing `Ctrl+Shift+P` (`Cmd+Shift+P` on macOS).
2. Type "Preferences: Color Theme" and select it from the list.
3. Scroll through the available themes and choose "Rypjaws" from the options.
4. The Rypjaws theme will be applied immediately.

## Website 
Please do visit our website for more information at [Rypjaws](https://rypjaws.netlify.app/)

## Feedback and Contributions

Feedback and contributions are always welcome! If you encounter any issues, have suggestions for improvements, or would like to contribute to the project, please visit the [GitHub repository](https://github.com/MihirRajeshPanchal/rypjaws) for the Rypjaws VS Code Theme Extension.

## License

This extension is released under the [MIT License](LICENSE).

## Credits

The Rypjaws theme was created by [Mihir Panchal](https://github.com/MihirRajeshPanchal). Special thanks to the following contributors for their valuable input and support.

<p align="start">
<a  href="https://github.com/MihirRajeshPanchal/rypjaws/graphs/contributors">
  <img src="https://contrib.rocks/image?repo=MihirRajeshPanchal/rypjaws"/>
</a>
</p>
Thank you for choosing the Rypjaws VS Code Theme Extension! We hope you enjoy coding with this stylish and immersive theme. If you find it useful, please consider sharing it with your friends and colleagues. Happy coding!


CHANGELOG.md
# Change Log

All notable changes to the "rypjaws" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release


rypjaws/src/components/Hero.js
import {
    Container,
    Stack,
    Flex,
    Box,
    Heading,
    Text,
    Button,
    Image,
    Icon,
    IconButton,
    IconProps,
    useColorModeValue,
  } from '@chakra-ui/react';
  import Output from './../assets/Output.png';
  
  export default function Hero() {
    return (
      <Container maxW={'7xl'}>
        <Stack
          align={'center'}
          spacing={{ base: 8, md: 10 }}
          py={{ base: 20, md: 28 }}
          direction={{ base: 'column', md: 'row' }}>
          <Stack flex={1} spacing={{ base: 5, md: 10 }}>
            <Heading
              lineHeight={1.1}
              fontWeight={600}
              fontSize={{ base: '3xl', sm: '4xl', lg: '6xl' }}>
              <Text
                as={'span'}
                position={'relative'}
                _after={{
                  content: "''",
                  width: 'full',
                  height: '30%',
                  position: 'absolute',
                  bottom: 1,
                  left: 0,
                  bg: 'rgba(86,153,0,255)',
                  zIndex: -1,
                }}>
              </Text>
              RypJaws
              <br />
              <Text as={'span'} color={'rgba(86,153,0,255)'}>
              </Text>
            </Heading>
            <Text color={'gray.500'}>
              Welcome to the Rypjaws VS Code Theme Extension! This extension brings the Rypjaws theme to your Visual Studio Code editor, providing a stylish and immersive coding experience.
            </Text>
            <Stack
              spacing={{ base: 4, sm: 6 }}
              direction={{ base: 'column', sm: 'row' }}>
              <Button
                rounded={'full'}
                size={'lg'}
                fontWeight={'normal'}
                px={6}
                colorScheme={'lime'}
                bg={'rgba(86,153,0,255)'}
                _hover={{ bg: 'rgba(86,153,0,255)' }}
                style={{ boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.1)' }}>
                <a href="vscode:extension/MihirPanchal.rypjaws">
                  Get started
                </a>
              </Button>
              <Button
                rounded={'full'}
                size={'lg'}
                fontWeight={'normal'}
                px={6}
                style={{ boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.1)' }}>
                <a href="https://marketplace.visualstudio.com/items?itemName=MihirPanchal.rypjaws">
                  How It Works
                </a>
              </Button>
            </Stack>
          </Stack>
          <Flex
            flex={1}
            justify={'center'}
            align={'center'}
            position={'relative'}
            w={'full'}>
            <Blob
              w={'150%'}
              h={'150%'}
              position={'absolute'}
              top={'-20%'}
              left={0}
              zIndex={-1}
              color={useColorModeValue('rgba(86,153,0,255)', 'rgba(86,153,0,255)')}
            />
            <Box
              position={'relative'}
              height={'300px'}
              rounded={'2xl'}
              boxShadow={'2xl'}
              width={'full'}
              overflow={'hidden'}>
              <IconButton
                aria-label={'Play Button'}
                variant={'ghost'}
                _hover={{ bg: 'transparent' }}
                size={'lg'}
                color={'white'}
                position={'absolute'}
                left={'50%'}
                top={'50%'}
                transform={'translateX(-50%) translateY(-50%)'}
              />
              <Image
                alt={'Hero Image'}
                fit={'cover'}
                align={'center'}
                w={'100%'}
                h={'100%'}
                src={Output}
              />
            </Box>
          </Flex>
        </Stack>
      </Container>
    );
  }
  
  export const Blob = (props: IconProps) => {
    return (
      <Icon
        width={'100%'}
        viewBox="0 0 578 440"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
        {...props}>
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M239.184 439.443c-55.13-5.419-110.241-21.365-151.074-58.767C42.307 338.722-7.478 282.729.938 221.217c8.433-61.644 78.896-91.048 126.871-130.712 34.337-28.388 70.198-51.348 112.004-66.78C282.34 8.024 325.382-3.369 370.518.904c54.019 5.115 112.774 10.886 150.881 49.482 39.916 40.427 49.421 100.753 53.385 157.402 4.13 59.015 11.255 128.44-30.444 170.44-41.383 41.683-111.6 19.106-169.213 30.663-46.68 9.364-88.56 35.21-135.943 30.551z"
          fill="currentColor"
        />
      </Icon>
    );
  };
  

rypjaws/src/components/Footer.js
import {
    Box,
    chakra,
    Container,
    Link,
    Stack,
    Text,
    Image,
    useColorModeValue,
    VisuallyHidden,
  } from '@chakra-ui/react';
  import { FaInstagram, FaTwitter, FaGithub, FaLinkedin } from 'react-icons/fa';
  import { ReactNode } from 'react';
  import LogoWhite from "./../assets/RypJaws-white.png"
  
  const SocialButton = ({
    children,
    label,
    href,
  }: {
    children: ReactNode;
    label: string;
    href: string;
  }) => {
    return (
      <chakra.button
        bg={useColorModeValue('blackAlpha.100', 'whiteAlpha.100')}
        rounded={'full'}
        w={8}
        h={8}
        cursor={'pointer'}
        as={'a'}
        href={href}
        display={'inline-flex'}
        alignItems={'center'}
        justifyContent={'center'}
        transition={'background 0.3s ease'}
        _hover={{
          bg: useColorModeValue('blackAlpha.200', 'whiteAlpha.200'),
        }}>
        <VisuallyHidden>{label}</VisuallyHidden>
        {children}
      </chakra.button>
    );
  };
  
  export default function SmallCentered() {
    return (
      <Box
        bg={useColorModeValue('gray.50', 'gray.900')}
        color={useColorModeValue('gray.700', 'gray.200')}>
        <Container
          as={Stack}
          maxW={'6xl'}
          py={4}
          spacing={4}
          justify={'center'}
          align={'center'}>
          <Box>
              <Image src={LogoWhite} boxSize="60px" />
            </Box>
          <Stack direction={'row'} spacing={6}>
            <Link href={'#'}>Home</Link>
            <Link href={'https://github.com/MihirRajeshPanchal/rypjaws'}>Repository</Link>
            <Link href={'https://marketplace.visualstudio.com/items?itemName=MihirPanchal.rypjaws'}>Marketplace</Link>
            <Link href={'vscode:extension/MihirPanchal.rypjaws'}>Downloads</Link>
            <Link href={'https://github.com/MihirRajeshPanchal/rypjaws/discussions'}>Community</Link>
            <Link href={'https://github.com/MihirRajeshPanchal/rypjaws/graphs/contributors'}>Team</Link>
          </Stack>
        </Container>
  
        <Box
          borderTopWidth={1}
          borderStyle={'solid'}
          borderColor={useColorModeValue('gray.200', 'gray.700')}>
          <Container
            as={Stack}
            maxW={'6xl'}
            py={4}
            direction={{ base: 'column', md: 'row' }}
            spacing={4}
            justify={{ base: 'center', md: 'space-between' }}
            align={{ base: 'center', md: 'center' }}>
            <Text>© 2023 Rypjaws. All rights reserved </Text>
            <Stack direction={'row'} spacing={6}>
              <SocialButton label={'Github'} href={'https://github.com/MihirRajeshPanchal'}>
                <FaGithub />
              </SocialButton>
              <SocialButton label={'LinkedIn'} href={'https://www.linkedin.com/in/mihir-panchal-9b6677220/'}>
                <FaLinkedin />
              </SocialButton>
              <SocialButton label={'Instagram'} href={'https://www.instagram.com/mihir_panchal_16/'}>
                <FaInstagram />
              </SocialButton>
              <SocialButton label={'Twitter'} href={'https://twitter.com/MihirPa81993010'}>
                <FaTwitter />
              </SocialButton>
            </Stack>
          </Container>
        </Box>
      </Box>
    );
  }

rypjaws/src/components/Navbar.js
import { ReactNode } from 'react';
import {
  Image,
  Box,
  Flex,
  HStack,
  Link,
  IconButton,
  Button,
  useDisclosure,
  useColorModeValue,
  Stack,
} from '@chakra-ui/react';
import { HamburgerIcon, CloseIcon, StarIcon } from '@chakra-ui/icons';
import Logo from "./../assets/RypJaws-white.png";

const Links = [
  { label: 'Repository', href: 'https://github.com/MihirRajeshPanchal/rypjaws' },
  { label: 'Marketplace', href: 'https://marketplace.visualstudio.com/items?itemName=MihirPanchal.rypjaws' },
  { label: 'Downloads', href: 'vscode:extension/MihirPanchal.rypjaws' },
  { label: 'Community', href: 'https://github.com/MihirRajeshPanchal/rypjaws/discussions' },
  { label: 'Team', href: 'https://github.com/MihirRajeshPanchal/rypjaws/graphs/contributors' },
];

const NavLink = ({ children, href }: { children: ReactNode, href: string }) => (
  <Link
    px={2}
    py={1}
    rounded={'md'}
    _hover={{
      textDecoration: 'none',
      bg: useColorModeValue('gray.200', 'gray.700'),
    }}
    href={href}>
    {children}
  </Link>
);

export default function Navbar() {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <>
      <Box bg={useColorModeValue('gray.100', 'gray.900')} px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <IconButton
            size={'md'}
            icon={isOpen ? <CloseIcon /> : <HamburgerIcon />}
            aria-label={'Open Menu'}
            display={{ md: 'none' }}
            onClick={isOpen ? onClose : onOpen}
          />
          <HStack spacing={8} alignItems={'center'}>
            <Box>
              <Image src={Logo} boxSize="60px" />
            </Box>
            <HStack as={'nav'} spacing={4} display={{ base: 'none', md: 'flex' }}>
              {Links.map((link) => (
                <NavLink key={link.label} href={link.href}>
                  {link.label}
                </NavLink>
              ))}
            </HStack>
          </HStack>
          <Flex alignItems={'center'}>
            <Button
              variant={'solid'}
              colorScheme={'green'}
              size={'sm'}
              mr={4}
              leftIcon={<StarIcon />}
            >
              <a href="https://github.com/MihirRajeshPanchal/rypjaws">
                Star on Github
              </a>
            </Button>
          </Flex>
        </Flex>

        {isOpen ? (
          <Box pb={4} display={{ md: 'none' }}>
            <Stack as={'nav'} spacing={4}>
              {Links.map((link) => (
                <NavLink key={link.label} href={link.href}>
                  {link.label}
                </NavLink>
              ))}
            </Stack>
          </Box>
        ) : null}
      </Box>
    </>
  );
}


rypjaws/src/components/Installation.js
import { Box, Heading, Flex, Text, VStack } from "@chakra-ui/react";
import { CheckCircleIcon } from "@chakra-ui/icons";

const InstallationSteps = () => {
    return (
      <Flex justify="center" align="center" height="100vh">
        <Box p={4}>
          <Heading as="h2" size="lg" mb={4} textAlign="center">
            Installation Steps
          </Heading>
          <VStack spacing={4} align="start">
            <InstallationStepCard
              step="Step 1"
              description="Launch Visual Studio Code."
            />
            <InstallationStepCard
              step="Step 2"
              description="Open the Extensions view by clicking on the square icon in the left sidebar or by pressing Ctrl+Shift+X (Cmd+Shift+X on macOS)."
            />
            <InstallationStepCard
              step="Step 3"
              description="Search for 'Rypjaws' in the Extensions Marketplace."
            />
            <InstallationStepCard
              step="Step 4"
              description="Click on the 'Install' button for the Rypjaws VS Code Theme Extension."
            />
            <InstallationStepCard
              step="Step 5"
              description="Once installed, click on the 'Reload' button to activate the theme."
            />
          </VStack>
        </Box>
      </Flex>
    );
  };
  

const InstallationStepCard = ({ step, description }) => {
  return (
    <Box
      bg="white"
      shadow="md"
      rounded="md"
      p={4}
      width="100%"
      borderWidth="1px"
      borderColor="gray.200"
      display="flex"
      alignItems="center"
      fontSize="lg"
      _hover={{
        bg: "gray.100",
      }}
    >
      <Box
        mr={2}
        color="green.500"
        boxSize={6}
        as={CheckCircleIcon}
      />
      <Flex direction="column">
        <Text fontWeight="bold" fontSize="xl" mb={2}>
          {step}
        </Text>
        <Text>{description}</Text>
      </Flex>
    </Box>
  );
};

export default InstallationSteps;


rypjaws/src/components/Features.js
import { Box, Container, Flex, Heading, Icon, Stack, Text, useColorModeValue } from '@chakra-ui/react';
import { ReactElement, useEffect, useRef } from 'react';
import { motion, useAnimation } from 'framer-motion';
import { FcBriefcase, FcTreeStructure, FcLineChart, FcMultipleDevices, FcBullish } from 'react-icons/fc';

interface CardProps {
  heading: string;
  description: string;
  icon: ReactElement;
  href: string;
}

const Card = ({ heading, description, icon, href }: CardProps) => {
  const controls = useAnimation();
  const cardRef = useRef(null);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const handleIntersection = (entries) => {
    const [entry] = entries;
    if (entry.isIntersecting) {
      controls.start({ scale: 1 });
    }
  };

  useEffect(() => {
    const observer = new IntersectionObserver(handleIntersection);
    if (cardRef.current) {
      observer.observe(cardRef.current);
    }

    return () => {
      observer.disconnect();
    };
  }, [handleIntersection]);

  return (
    <motion.div
      ref={cardRef}
      initial={{ scale: 0 }}
      animate={controls}
      transition={{ duration: 0.4 }}
    >
      <Box
        maxW={{ base: 'full', md: '275px' }}
        w={'full'}
        borderWidth="1px"
        borderRadius="lg"
        overflow="hidden"
        p={5}
        style={{
          boxShadow: '0 4px 6px rgba(0, 0, 0, 0.1)',
          transition: 'box-shadow 0.2s ease-in-out',
        }}
        whileHover={{ boxShadow: '0 6px 8px rgba(0, 0, 0, 0.2)' }}
      >
        <Stack align={'start'} spacing={2}>
          <Flex
            w={16}
            h={16}
            align={'center'}
            justify={'center'}
            color={'white'}
            rounded={'full'}
            bg={useColorModeValue('gray.100', 'gray.700')}
          >
            {icon}
          </Flex>
          <Box mt={2}>
            <Heading size="md">{heading}</Heading>
            <Text mt={1} fontSize={'sm'}>
              {description}
            </Text>
          </Box>
        </Stack>
      </Box>
    </motion.div>
  );
};

export default function gridListWith() {
  return (
    <Box p={4}>
      <Stack spacing={4} as={Container} maxW={'3xl'} textAlign={'center'}>
        <Heading fontSize={{ base: '2xl', sm: '4xl' }} fontWeight={'bold'}>
          Features
        </Heading>
        <Text color={'gray.600'} fontSize={{ base: 'sm', sm: 'lg' }}>
          RypJaws is a free and open-source extension that is available for download from the Visual Studio Code Marketplace.
        </Text>
      </Stack>

      <Container maxW={'5xl'} mt={12}>
        <Flex flexWrap="wrap" gridGap={6} justify="center">
          <Card
            heading={'Elegant Color Scheme'}
            icon={<Icon as={FcTreeStructure} w={10} h={10} />}
            description={
              'Rypjaws theme offers a carefully chosen set of colors that are easy on the eyes and enhance your focus on the code.'
            }
            href={'#'}
          />
          <Card
            heading={'Syntax Highlighting'}
            icon={<Icon as={FcLineChart} w={10} h={10} />}
            description={
              'The theme provides syntax highlighting for various programming languages, making your code stand out and improving readability.'
            }
            href={'#'}
          />
          <Card
            heading={'Consistent Design'}
            icon={<Icon as={FcMultipleDevices} w={10} h={10} />}
            description={
              'Rypjaws maintains a consistent design across different elements of the editor, ensuring a cohesive and visually pleasing experience'
            }
            href={'#'}
          />
          <Card
            heading={'Customized UI Components'}
            icon={<Icon as={FcBullish} w={10} h={10} />}
            description={
              'The extension modifies the appearance of various UI components, including the sidebar, activity bar, status bar, and more, to match the overall aesthetic of the theme'
            }
            href={'#'}
          />
          <Card
            heading={'Optimized for Productivity'}
            icon={<Icon as={FcBriefcase} w={10} h={10} />}
            description={
              'Rypjaws is designed to help you stay productive with its carefully selected colors and contrast ratios, allowing you to work for long hours without straining your eyes.'
            }
            href={'#'}
          />
        </Flex>
      </Container>
    </Box>
  );
}
